build.sh

```diff
@@ -1,49 +1,62 @@
-#!/bin/bash
+#! /usr/bin/env bash
 
 set -e # exit if a command fails
 set -o pipefail # Will return the exit status of make if it fails
 
-CFLAGS_COMMON="-O0 -g -DXBOX=1" # Compilation flags for all platforms
-POST_BUILD=""
-
 package_windows() { # Script to prepare the windows exe
-    mkdir -p dist
-    cp i386-softmmu/qemu-system-i386.exe dist/xqemu.exe
-    cp i386-softmmu/qemu-system-i386w.exe dist/xqemuw.exe
-    python2 ./get_deps.py dist/xqemu.exe dist
-    strip dist/xqemu.exe
-    strip dist/xqemuw.exe
+	mkdir -p dist
+	cp i386-softmmu/qemu-system-i386.exe dist/xqemu.exe
+	cp i386-softmmu/qemu-system-i386w.exe dist/xqemuw.exe
+	python2 ./get_deps.py dist/xqemu.exe dist
+	strip dist/xqemu.exe
+	strip dist/xqemuw.exe
 }
 
+postbuild=''
+build_cflags='-O3'
+job_count='4'
+
+if [ "${1:0:2}" = '-j' ]
+then
+	job_count="${1:2}"
+	shift
+fi
+
+if echo " ${@} " | grep -q '[[:space:]]--enable-debug[[:space:]]'
+then
+	build_cflags='-O0 -g'
+fi
+
 case "$(uname -s)" in # adjust compilation option based on platform
-    Linux)
-        echo "Compiling for Linux..."
-        CFLAGS="-march=native -Wno-error=redundant-decls -Wno-error=unused-but-set-variable"
-        CONFIGURE="--enable-kvm --disable-xen --disable-werror"
-        ;;
-    Darwin)
-        echo "Compiling for MacOS..."
-        CFLAGS="-march=native"
-        CONFIGURE="--disable-cocoa"
-        ;;
-    CYGWIN*|MINGW*|MSYS*)
-        echo "Compiling for Windows..."
-        CFLAGS="-Wno-error"
-        CONFIGURE="--python=python2 --disable-cocoa --disable-opengl"
-        POST_BUILD="package_windows" # set the above function to be called after build
-        ;;
-    *)
-        echo "Could not detect OS $(uname -s), aborting."
-        exit -1
-        ;;
+	Linux)
+		echo 'Compiling for Linux…'
+		sys_cflags='-march=native -Wno-error=redundant-decls -Wno-error=unused-but-set-variable'
+		sys_opts='--enable-kvm --disable-xen --disable-werror'
+		;;
+	Darwin)
+		echo 'Compiling for MacOS…'
+		sys_cflags='-march=native'
+		sys_opts='--disable-cocoa'
+		;;
+	CYGWIN*|MINGW*|MSYS*)
+		echo 'Compiling for Windows…'
+		sys_cflags='-Wno-error'
+		sys_opts='--python=python2 --disable-cocoa --disable-opengl'
+		postbuild='package_windows' # set the above function to be called after build
+		;;
+	*)
+		echo "could not detect OS $(uname -s), aborting" >&2
+		exit -1
+		;;
 esac
 
+configure="$(dirname "$(readlink -f "${0}")")/configure"
```

Comment:
Probably want to move this to move the generated `configure` directory path and explain why this is so complicated in a comment